Essentials/src/main/java/com/earth2me/essentials/IUser.java

```diff
@@ -223,4 +235,74 @@
     void setToggleShout(boolean toggleShout);
 
     boolean isToggleShout();
+
+    /**
+     * Gets information about the last-made tpa request in the tpa queue of this {@link IUser}.
+     *
+     * The TPA Queue is Last-In-First-Out queue which stores all the active pending teleport
+     * requests of this {@link IUser}. Timeout calculations are also done during the
+     * iteration process of this method, ensuring that teleport requests made past the timeout
+     * period are removed from queue and therefore not returned here. The maximum size of this
+     * queue is determined by {@link ISettings#getTpaMaxAmount()}.
+     *
+     * @param inform      true if the underlying {@link IUser} should be informed if a request expires during iteration.
+     * @param shallow     true if this method should not spend time validating time for all items in the queue and just return the first item in the queue.
+     * @param excludeHere true if /tphere requests should be ignored in fetching the next tpa token.
+     * @return A {@link TpaRequest} corresponding to the next available request or null if no valid request is present.
+     */
+    TpaRequest getNextTpaToken(boolean inform, boolean shallow, boolean excludeHere);
+
+    /**
+     * Weather or not this {@link IUser} has any valid TPA request in queue.
+     *
+     * @param inform      true if the underlying {@link IUser} should be informed if a request expires during iteration.
+     * @param excludeHere true if /tphere requests should be ignored in checking if a tpa request is available.
```

Comment:
```suggestion
     * @param excludeHere true if /tpahere requests should be ignored in checking if a tpa request is available.
```

`/tphere` is the non-request version, `/tpahere` is the request command.